workflow/executor/executor.go

```diff
@@ -980,9 +980,11 @@ func (we *WorkflowExecutor) monitorProgress(ctx context.Context, progressFile st
 			log.WithError(ctx.Err()).Info("stopping progress monitor (context done)")
 			return
 		case <-annotationPatchTicker.C:
-			log.WithField("progress", we.progress).Infof("patching pod progress annotation")
-			if err := we.AddAnnotation(ctx, common.AnnotationKeyProgress, we.progress); err != nil {
-				log.WithField("progress", we.progress).WithError(err).Warn("failed to patch progress annotation")
+			if we.progress.IsValid() {
```

Comment:
... else print warning?

Reply:
I just saw #7184. Perhaps then

```go
if we.progress.IsValid() {
  ...
} else if we.progress != "" {
  log.Warnf("invalid progress...") // This is just an example, actual log with correct parameters of course
}
```

Reply:
I too did wonder if I should print a warning or not. I didn't add one because `progress` at this point should only be invalid when no progress is reported by the pod. In this case you're probably not interested in the warning, unless you're debugging something related to this. If the progress could not be parsed (invalid format) that's reported elsewhere already. 

So, should I add a warning or not?

Reply:
> I didn't add one because progress at this point should only be invalid when no progress is reported by the pod.

Then perhaps 
```suggestion
			if we.progress != "" {
```
is more explicit?

> If the progress could not be parsed (invalid format) that's reported elsewhere already.

Great, then no warning necessary.

Reply:
fine for me 👍 changed the line ✅